common/image_u8.c

```diff
@@ -321,29 +321,29 @@ void image_u8_convolve_2D(image_u8_t *im, const uint8_t *k, int ksz)
 {
     assert((ksz & 1) == 1); // ksz must be odd.
 
+    uint8_t *x = malloc(sizeof(uint8_t)*im->stride);
     for (int y = 0; y < im->height; y++) {
 
-        uint8_t *x = malloc(sizeof(uint8_t)*im->stride);
         memcpy(x, &im->buf[y*im->stride], im->stride);
 
         convolve(x, &im->buf[y*im->stride], im->width, k, ksz);
-        free(x);
     }
+    free(x);
 
+    uint8_t *xb = malloc(sizeof(uint8_t)*im->height);
+    uint8_t *yb = malloc(sizeof(uint8_t)*im->height);
     for (int x = 0; x < im->width; x++) {
-        uint8_t *xb = malloc(sizeof(uint8_t)*im->height);
-        uint8_t *yb = malloc(sizeof(uint8_t)*im->height);
 
         for (int y = 0; y < im->height; y++)
             xb[y] = im->buf[y*im->stride + x];
 
         convolve(xb, yb, im->height, k, ksz);
-        free(xb);
 
         for (int y = 0; y < im->height; y++)
             im->buf[y*im->stride + x] = yb[y];
-        free(yb);
     }
+    free(xb);
+    free(yb);
```

Comment:
This looks unrelated to the implementation of the parallel method and looks more like a performance improvement by reducing the number of memory allocations. If so, add this in a dedicated commit.

Reply:
If you really want. Otherwise, since this change is quite trivial, I would prefer to just have a separate commit in this very PR.